ores/scoring_systems/celery_queue.py

```diff
@@ -0,0 +1,287 @@
+import logging
+import re
+from urllib.parse import urlparse
+from itertools import chain
+
+import celery
+import celery.exceptions
+import mwapi.errors
+import revscoring.errors
+from celery.signals import before_task_publish
+
+from ..errors import TimeoutError, ScoreProcessorOverloaded
+from .scoring_system import ScoringSystem
+
+logger = logging.getLogger(__name__)
+
+APPLICATIONS = []
+
+DEFAULT_CELERY_QUEUE = "celery"
+
+
+@before_task_publish.connect
+def update_sent_state(sender=None, body=None, **kwargs):
+
+    for application in APPLICATIONS:
+        task = application.tasks.get(sender)
+        backend = task.backend if task else application.backend
+
+        logger.debug("Setting state to 'SENT' for {0}".format(body['id']))
+        backend.store_result(body['id'], result=None, status="SENT")
+
+
+class CeleryQueue(ScoringSystem):
+
+    def __init__(self, *args, application, queue_maxsize=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.application = application
+        self.queue_maxsize = int(queue_maxsize) if queue_maxsize is not None \
+                             else None
+
+        self.redis = redis_from_url(self.application.conf.BROKER_URL)
+
+        if self.queue_maxsize is not None and self.redis is None:
+            logger.warning("No redis connection.  Can't check queue size")
+
+        expected_errors = (revscoring.errors.RevisionNotFound,
+                           revscoring.errors.PageNotFound,
+                           revscoring.errors.UserNotFound,
+                           revscoring.errors.DependencyError,
+                           mwapi.errors.RequestError,
+                           mwapi.errors.TimeoutError,
+                           TimeoutError)
```

Comment:
Since there's a name collision already, it would be helpful if the second `TimeoutError` was fully qualified